File: android/resourceextensions/src/main/java/com/shopify/resourceextensions/PropertiesFactory.kt
package com.shopify.resourceextensions

class PropertiesFactory : Factory {

    override fun build(context: Context, args: kotlin.collections.List<Node>): Any {
        val map = HashMap<String, Any?>()
        for (x in 0 until args.count() step 2) {
            val key = args[x].evaluate(context) as String
            val value = args[x+1].evaluate(context)
            map[key] = value
        }
        return Properties(map)
    }

}

class Properties(private val map: Map<String, Any?>) {

    operator fun get(key: String) = map[key]

}

File: ios/AppExtensionLib/AppExtensionLib/PropertiesFactory.swift
//
//  PropertiesFactory.swift
//  AppExtensionLib
//
//  Created by Dave Pelletier on 2019-10-30.
//  Copyright © 2019 Shopify. All rights reserved.
//

import Foundation

class PropertiesFactory : Factory {
    
    func build(context: Context, args: [Any?]) -> Any {
        var map = [String: Any]()
        for i in stride(from: 0, to: args.count, by: 2) {
            let key = args[i] as! String
            let value = args[i+1]
            map[key] = value
        }
        return map
    }
    
}
